### PR TITLE
Run naming styles on closed files

### LINKED ISSUE #15760: The Error List shows different results based on whether you've ever opened a source file
1. Clone this repo to this commit: https://github.com/dotnet/roslyn-project-system/commit/e8f444a1ebe3d78b6b494d5dc355e95b7da19086
2. From a VS command prompt, run build.cmd from the root
3. Open src\ProjectSystem.sln (build 26005.0 D15PreRel)
4. Change your VB code style settings to:
image
6. Make sure Background Analysis is turned on for VB
7. Wait till Error List no longer has "..." after name
-- Error List should look something like --
image

7. Open src\Microsoft.VisualStudio.AppDesigner\ApplicationDesigner\ApplicationDesignerView.vb
8. Close that file

-- Error List should look something like (Notice that there's 22 more messages that have appeared) -- 

image

Very confused how or why messages show up the Error List? What exactly do I need to do to show *all* possible messages?

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Our best guess is we ran naming styles on open files only for two reasons:

1. At the time, we only supported .editorconfig in-proc, so out of proc wouldn't have used the right settings.
2. Even if we were running in-proc, there might have been performance issues since we were running across a lot of files.

Neither of these concerns apply at this point: we now support .editorconfig out of proc (both in the legacy path and the new path) and since we're already running other analyzers that use .editorconfig,
the cost of processing it will exist regardless.

Fixes #15760

### CODE REVIEW

File: src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -41,8 +41,6 @@ protected NamingStyleDiagnosticAnalyzerBase()
         // see https://github.com/dotnet/roslyn/issues/14061
         protected abstract ImmutableArray<TLanguageKindEnum> SupportedSyntaxKinds { get; }
 
-        public override bool OpenFileOnly(Workspace workspace) => true;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs]: Hooray!

File: src/Workspaces/Core/Portable/Execution/AbstractOptionsSerializationService.cs [hunk 2]
```diff
@@ -316,30 +316,6 @@ private OptionSet ReadOptionFrom<T>(OptionSet options, string language, PerLangu
             return options.WithChangedOption(option, language, value);
         }
 
-        protected void WriteOptionTo(OptionSet options, Option<NamingStylePreferences> option, ObjectWriter writer, CancellationToken cancellationToken)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Execution/AbstractOptionsSerializationService.cs]: so these were just not called?

Comment (jasonmalinowski) [reply src/Workspaces/Core/Portable/Execution/AbstractOptionsSerializationService.cs]: Nope.

**Comment by mavasani** [comment]
@jasonmalinowski Any progress on this one? There is a VS feedback [user request](https://github.com/dotnet/roslyn/issues/38051#issuecomment-545665674) to view all naming violations in a project/solution which depends on this PR.

**Comment by mavasani** [comment]
Another VS feedback requesting naming violations on closed files: https://developercommunity.visualstudio.com/content/problem/880060/vs2019-naming-rules-applied-inconsistently.html

@jasonmalinowski Any update on this PR?

**Comment by jasonmalinowski** [comment]
RPS run passed internally, which is what @sharwell suspected since this is unlikely running in that scenario in the first place.

**Comment by jasonmalinowski** [comment]
@mavasani and @sharwell any concerns prior to merge? Your signoffs are ancient at this point but nothing really changed.

**Comment by mavasani** [comment]
Nope, looks all good to merge. Thanks Jason!

**Comment by davkean** [comment]
OMG - thank-you thank-you thank-you!

**Comment by sharwell** [comment]
So glad this made the cutoff!

**Comment by mavasani** [comment]
You should also be able to get advantage of this feature with "Open Document" analysis scope + invoking "Run Code Analysis" command on project/solution for an demand explicit execution.

**Pull request merged**